stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -274,6 +274,7 @@ namespace stan {
        * Iain Murray: Differentiation of the Cholesky decomposition, 2016.
        *
        */
+    #if STAN_GPU
```

Comment:
Should the whole `cholesky_gpu` class be inside this `#if STAN_GPU` block? Or is that class sometimes used when we aren't using a GPU? That would seem a bit odd at least from a naming perspective, though I might be missing something.